EXILED/Exiled.API/Features/Items/Jailbird.cs

```diff
@@ -114,12 +117,44 @@ public JailbirdWearState WearState
             get => Base._deterioration.WearState;
             set
             {
-                if (JailbirdDeteriorationTracker.ReceivedStates.ContainsKey(Serial))
-                    JailbirdDeteriorationTracker.ReceivedStates[Serial] = value;
+                TotalDamageDealt = GetDamage(value);
+                TotalCharges = GetCharge(value);
                 Base._deterioration.RecheckUsage();
             }
         }
 
+        /// <summary>
+        /// Calculates the damage corresponding to a given <see cref="JailbirdWearState"/>.
+        /// </summary>
+        /// <param name="wearState">The wear state to calculate damage for.</param>
+        /// <returns>The amount of damage associated with the specified wear state.</returns>
+        public float GetDamage(JailbirdWearState wearState)
+        {
+            foreach (Keyframe keyframe in Base._deterioration._damageToWearState.keys)
+            {
+                if (Base._deterioration.FloatToState(keyframe.value) == wearState)
+                    return keyframe.time;
+            }
+
+            throw new Exception("Wear state not found in charges to wear state mapping.");
+        }
+
+        /// <summary>
+        /// Gets the charge needed to reach a specific <see cref="JailbirdWearState"/>.
+        /// </summary>
+        /// <param name="wearState">The desired wear state to calculate the charge for.</param>
+        /// <returns>The charge value required to achieve the specified wear state.</returns>
+        public int GetCharge(JailbirdWearState wearState)
```

Comment:
It will not work

Reply:
> It will not work

Why that how nw does it ?

Reply:
https://github.com/Exiled-Team/EXILED/pull/2736

Reply:
Resolved